base/cvd/cuttlefish/common/libs/utils/files.cpp

```diff
@@ -81,6 +81,30 @@ bool FileExists(const std::string& path, bool follow_symlinks) {
   return (follow_symlinks ? stat : lstat)(path.c_str(), &st) == 0;
 }
 
+Result<dev_t> FileDeviceId(const std::string& path) {
+  struct stat out;
+  CF_EXPECTF(stat(path.c_str(), &out) == 0,
+             "stat() failed trying to retrive device ID information for \"{}\" "
+             "with error: {}",
+             path, strerror(errno));
+  return out.st_dev;
+}
+
+Result<bool> CanHardLink(const std::string& path1, const std::string& path2) {
```

Comment:
Rename the parameters to make it clear one is the target and the other one the parent directory of the link that can be created.

Reply:
This helper is only checking if the two paths can be hardlinked, there are no particular semantics for which is which.  The parameters are distinct for `CreateHardLink`, but here we are just calling `stat` and checking both have the same device ids.  They could both be any type of file.